client/jetpack-onboarding/main.jsx

```diff
@@ -145,7 +145,7 @@ class JetpackOnboardingMain extends React.PureComponent {
 				   * querying data for the given site. The `jpoAuth` connected prop depends on whether
 				   * the site is a connected Jetpack site or not, and a network request that uses
 				   * the wrong argument can mess up our request tracking quite badly. */
-				this.state.hasFinishedRequestingSite && (
+				( this.state.hasFinishedRequestingSite || jpoAuth ) && (
```

Comment:
This goes totally against the comment above 🙁  IIRC, there was a bug that was due to a different `jpoAuth` query used here, and in https://github.com/Automattic/wp-calypso/blob/39844e22b0463bf16471c8dab80f9dbfda78fd78/client/jetpack-onboarding/main.jsx#L209-L212: When Calypso finished fetching site data and the site turned out to be connected, `jpoAuth` would change to `null` (or `undefined`, doesn't matter), and `isRequestingSettings` would then check for a pending request using that `null` query. Obviously, there would be no such pending query, so it's falsely return `false`, causing the component to think we're done fetching site settings.

It's possible that the `|| jpoAuth` check here covers that case, but I'll try to find the original issue since we should definitely ensure we're not introducing a regression here. If we're going with the present approach, we should definitely update the comment tho, since this is pretty subtle.

Reply:
Thinking about this some more, the change is probably fine to make here (as opposed to `isRequestingSettings`, which would be hairier). However, we should really modify the comment.

Reply:
Let's see if #23472 addresses this one.